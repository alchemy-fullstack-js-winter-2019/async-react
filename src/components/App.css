.header {
  text-align: center;
  font-family: 'Open Sans', sans-serif;
  color: #555;
  text-decoration: none;
  text-transform: uppercase;
  font-size: 125px;
  font-weight: 800;
  letter-spacing: -3px;
  line-height: 1;
  text-shadow: #EDEDED 3px 2px 0;
  position: relative;
  margin-bottom: 50px;
}

.controls {
  text-align: center;
}

.locationControl {
  position: absolute;
  top: 245px;
  right: 400px;

}