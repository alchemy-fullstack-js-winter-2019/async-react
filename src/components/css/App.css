.pageNext {
  width: 300px;
}

.header {
  padding: 30px;
  background-color: grey;
  color: white;
  width: 100%;
}

.pageContainer {
  display: flex;
}
