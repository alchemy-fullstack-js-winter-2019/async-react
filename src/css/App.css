h1 {
  text-align: center;
  font-size: 4em;
  margin: 0;
  text-shadow: 2px 2px blue;
  padding: 10px;
  background: black;
  color: white;
}

ul {
  display: grid;
  grid-template-columns: repeat(4, 1fr);
  list-style-type: none;
  padding: 0;
  grid-column-gap: 10px;
  grid-row-gap: 10px;
}

li {
  list-style: none;
  position: relative;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: space-evenly;
  border: 2px solid navy;
  box-shadow: 4px 4px 4px black;
  background: black;
  margin: 0px 10px;
  color: white;
}

.info {
  background: white;
  width: 100%;
  color: navy;
}

* {
  font-family: 'montserrat';
}