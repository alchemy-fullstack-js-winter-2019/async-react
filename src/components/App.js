import React, { Component } from 'react';
import Characters from './characters/Characters';
import Locations from './locations/Locations';
import './App.css';
import charStyles from './characters/Characters.css';

export default class App extends Component {
  render() {
    return (
      <>
        <h1>Rick and Mort, Yo!</h1>
        <div className={charStyles.characters}>
          <Characters />
        </div>
        <Locations />
      </>
    );
  }
}
