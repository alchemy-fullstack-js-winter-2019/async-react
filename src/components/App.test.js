import React from 'react';
import App from './App';
import renderer from 'react-test-renderer';

jest.mock('../services/rickAndMortyApi.js');

describe('App', () => {
  it('matches a snapshot', () => {
    const tree = renderer.create(
      <App />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

