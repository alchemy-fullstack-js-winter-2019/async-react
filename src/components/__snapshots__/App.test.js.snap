// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App matches a snapshot 1`] = `
Array [
  <h1>
    Rick and Mort, Yo!
  </h1>,
  <div
    className="characters"
  >
    <header>
      <p>
        Current Page: 
        1
      </p>
      <p>
        Total Pages: 
        1
      </p>
    </header>
    <button
      onClick={[Function]}
    >
      &lt;
    </button>
    <button
      onClick={[Function]}
    >
      &gt;
    </button>
    <ul />
  </div>,
  <header>
    <p>
      Current Page: 
      1
    </p>
    <p>
      Total Pages: 
      1
    </p>
  </header>,
  <button
    onClick={[Function]}
  >
    &lt;
  </button>,
  <button
    onClick={[Function]}
  >
    &gt;
  </button>,
  <ul />,
]
`;
